### PR TITLE
656 not all pool details are properly shown

### LINKED ISSUE #656: Not all Pool Details are Properly Shown
Why are these values not fetched properly?
Showing a question mark for theses values is not giving users comfort.

It's this pool, maybe the case for other pools as well:
https://app.midascapital.xyz/56/pool/1

Screenshot 2022-09-12 at 14 30 02

### PR DESCRIPTION
**Pull request by peetzweg**
+ upgrade react-query to newest version, which is a new package - https://github.com/tanstack/query
I thought this might fix the issue we had. But it did not. But let's go with the new version, worked fine so far. No breaking changes.

+ fixes #656 

+ minor refactors

### CODE REVIEW

File: packages/ui/hooks/fuse/useExtraPoolInfo.ts [hunk 1]
```diff
@@ -1,64 +1,55 @@
...
   const { midasSdk, currentChain, address } = useMidas();
 
-  const { data } = useQuery(
-    ['ExtraPoolInfo', currentChain.id, comptrollerAddress],
+  return useQuery(
+    ['useExtraPoolInfo', currentChain.id, comptrollerAddress],
     async () => {
-      if (comptrollerAddress) {
-        const comptroller = midasSdk.createComptroller(comptrollerAddress);
-        const oracle = midasSdk.getPriceOracle(await comptroller.callStatic.oracle());
```
Comment (peetzweg) [inline packages/ui/hooks/fuse/useExtraPoolInfo.ts:13]: This is what caused the error:
```
const oracle = midasSdk.getPriceOracle(await comptroller.callStatic.oracle());
```

If split up in two lines it works. Not sure why it just recently broke or why it does not work in the first place.

```
const oracleAddress = await comptroller.callStatic.oracle();
const oracle = midasSdk.getPriceOracle(oracleAddress);
```

As nothing relies on it later on, I just added it to the Promise Array in the line 35 in the changed file:

```
comptroller.callStatic
          .oracle()
          .then((oracleAddress) => midasSdk.getPriceOracle(oracleAddress)),
```

File: packages/ui/hooks/fuse/useExtraPoolInfo.ts [hunk 2]
```diff
@@ -1,64 +1,55 @@
...
-          pendingAdmin,
-        ] = await Promise.all([
-          midasSdk.contracts.FusePoolLensSecondary.callStatic.getPoolOwnership(comptrollerAddress),
-
-          comptroller.callStatic.closeFactorMantissa(),
-
-          comptroller.callStatic.liquidationIncentiveMantissa(),
-
-          // TODO wtf?
-          (() => {
```
Comment (carlomazzaferro) [inline packages/ui/hooks/fuse/useExtraPoolInfo.ts:29]: Yes, less WTFs is always good

File: packages/ui/hooks/useBorrowMinimum.ts [hunk 3]
```diff
@@ -34,8 +34,8 @@ export const useBorrowMinimum = (asset: FuseAsset) => {
       minBorrowUSD: Number(utils.formatUnits(response.data, 18)) * usdPrice,
       minBorrowNative: response.data,
       minBorrowAsset: response.data
-        .mul(utils.parseUnits('1', asset.underlyingDecimals))
-        .div(asset.underlyingPrice),
+        .div(asset.underlyingPrice)
+        .mul(utils.parseUnits('1', asset.underlyingDecimals)),
```
Comment (carlomazzaferro) [inline packages/ui/hooks/useBorrowMinimum.ts:38]: https://github.com/Midas-Protocol/monorepo/commit/ba7d5d0450a1dcea847ca6f2d79426408f669636

This change fixed the issue of the 0.35 showing as zero

Screenshot 2022-09-13 at 13 22 52

File: packages/ui/hooks/fuse/useExtraPoolInfo.ts (hunk 1 above)
Comment (carlomazzaferro) [reply packages/ui/hooks/fuse/useExtraPoolInfo.ts:13]: Interesting. Nice deugging and overall refactoring. LG, apart from the single comment

File: packages/ui/package.json [hunk 4]
```diff
@@ -22,6 +22,8 @@
     "@midas-capital/sdk": "workspace:*",
     "@midas-capital/types": "workspace:*",
     "@supabase/supabase-js": "^1.35.6",
+    "@tanstack/react-query": "^4.3.4",
+    "@tanstack/react-query-devtools": "^4.3.5",
```
[the remark this answers is not in the document]
Comment (smartdevcom) [reply packages/ui/package.json:26]: Is it re-branded?

**Review by smartdevcom — APPROVED** [review]

File: packages/ui/package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (peetzweg) [reply packages/ui/package.json:26]: yep, it's expanding into other js frontend frameworks as well.

**Pull request merged**